app/models/king.rb

```diff
@@ -1,7 +1,10 @@
 class King < Piece
 
   def valid_move?(x_destination, y_destination)
-    (cell_x - x_destination).abs <= 1 && (cell_y - y_destination).abs <= 1 && is_obstructed?([x_destination, y_destination])
+    return false unless (cell_x - x_destination).abs <= 1 && (cell_y - y_destination).abs <= 1  
+
+    found_piece = Piece.find_by(cell_x: x_destination, cell_y: y_destination, game_id: game_id)
+    (found_piece.nil? || found_piece.color != color) 
```

Comment:
I don't think you need these two lines at all. Were you checking to see if the piece had been rendered on the board? If so I have that covered in my board.js file